integration/data/test_ha.py

```diff
@@ -524,3 +531,159 @@ def test_expansion_with_rebuild(grpc_controller,  # NOQA
                  data4_len, random_string(data4_len))
     snap4 = Snapshot(dev, data4, address)
     snap4.verify_data()
+
+
+def test_expansion_rollback_with_rebuild(
+        grpc_controller, grpc_fixed_dir_replica1, grpc_fixed_dir_replica2):  # NOQA
+    """
+    The test flow:
+    1. Write random data into the block device.
+    2. Create the 1st snapshot.
+    3. Create an empty directory using the tmp meta file path of
+       the expansion disk for each replica.
+       This will fail the following expansion and trigger expansion rollback.
+    4. Try to expand the volume but fails. Then the automatic rollback will
+       be applied implicitly.
+    5. Check the volume status and if there are leftovers of
```

Comment:
Make sure to check the volume size as well.

Reply:
It's checked in `wait_for_volume_expansion()`